data-prepper-plugins/log-http-source/src/main/java/com/amazon/dataprepper/plugins/source/loghttp/LogHTTPService.java

```diff
@@ -0,0 +1,67 @@
+/*
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  The OpenSearch Contributors require contributions made to
+ *  this file be licensed under the Apache-2.0 license or a
+ *  compatible open source license.
+ *
+ *  Modifications Copyright OpenSearch Contributors. See
+ *  GitHub history for details.
+ */
+
+package com.amazon.dataprepper.plugins.source.loghttp;
+
+import com.amazon.dataprepper.model.buffer.Buffer;
+import com.amazon.dataprepper.model.record.Record;
+import com.amazon.dataprepper.plugins.source.loghttp.codec.JsonCodec;
+import com.linecorp.armeria.common.AggregatedHttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.server.annotation.Get;
+import com.linecorp.armeria.server.annotation.Post;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+
+public class LogHTTPService {
+
+    // TODO: support other data-types as request body, e.g. json_lines, msgpack
+    private final JsonCodec jsonCodec = new JsonCodec();
+    private final Buffer<Record<String>> buffer;
+    private final int bufferWriteTimeoutInMillis;
+
+    public LogHTTPService(int bufferWriteTimeoutInMillis, Buffer<Record<String>> buffer) {
+        this.buffer = buffer;
+        this.bufferWriteTimeoutInMillis = bufferWriteTimeoutInMillis;
+    }
+
+    @Get("/log/ingest")
+    public HttpResponse doGet(AggregatedHttpRequest aggregatedHttpRequest) {
+        return processRequest(aggregatedHttpRequest);
+    }
+
+    @Post("/log/ingest")
+    public HttpResponse doPost(AggregatedHttpRequest aggregatedHttpRequest) {
+        return processRequest(aggregatedHttpRequest);
+    }
+
+    private HttpResponse processRequest(AggregatedHttpRequest aggregatedHttpRequest) {
+        List<String> jsonList;
+        try {
+            jsonList = jsonCodec.parse(aggregatedHttpRequest.content());
+        } catch (IOException e) {
+            return HttpResponse.of(HttpStatus.BAD_REQUEST, MediaType.ANY_TYPE, "Bad request data format. Needs to be json array.");
+        }
+        for (String json: jsonList) {
+            try {
+                // TODO: switch to new API writeAll once ready
+                buffer.write(new Record<>(json), bufferWriteTimeoutInMillis);
```

Comment:
Do we want to write partial data into buffer ? If not, then the buffer write should take list of Records, acquire the number of permits equal to Record List size and then write to buffer else throw TimeoutException.


```tryAcquire(int permits, long timeout, TimeUnit unit)```

Reply:
So this change needs to happen at the buffer API level rather than here in the service handler. Thus I made the TODO comment. Right now it is accepting partial data, but it will be fixed in the buffer::writeAll.